cli/compose/convert/service.go

```diff
@@ -646,9 +665,12 @@ func convertDNSConfig(dns []string, dnsSearch []string) *swarm.DNSConfig {
 	return nil
 }
 
-func convertCredentialSpec(namespace Namespace, spec composetypes.CredentialSpecConfig, refs []*swarm.ConfigReference) (*swarm.CredentialSpec, error) {
+func convertCredentialSpec(namespace Namespace, spec *composetypes.CredentialSpecConfig, refs []*swarm.ConfigReference) (*swarm.CredentialSpec, error) {
 	var o []string
 
+	if spec == nil {
+		return nil, nil
```

Comment:
Although this is an internal function I don't think we should return a nil error and nil value here. 

```suggestion
		return nil, errors.New("spec is nil")
```